src/sdl/SDL/SDL.spec.ts

```diff
@@ -201,24 +193,34 @@ describe("SDL", () => {
       });
 
       it.each(fields)('should throw an error when credentials are missing "%s"', field => {
+        const yml = createSdlYml({
+          "services.web.credentials": { $set: omit(credentials, field) }
+        });
+
         expect(() => {
-          SDL.fromString(readBasicSdl({ credentials: omit(credentials, field) }), "beta3", "sandbox");
+          SDL.fromString(yml, "beta3", "sandbox");
         }).toThrowError(new SdlValidationError(`service "web" credentials missing "${field}"`));
       });
 
       it.each(fields)('should throw an error when credentials "%s" is empty', field => {
         credentials[field] = "";
+        const yml = createSdlYml({
+          "services.web.credentials": { $set: credentials }
```

Comment:
Pretty cool, I never used `immutability-helper` per se. Is it strongly typed?

Reply:
Yes. However dot-object is not. I still used it as it's for tests and it simplified code quite a bit. If we feel like it we could add own typing there.

Reply:
Immutability helper is generally used to ui state management :) but it's useful in this case it seems

Reply:
Yea, I used immutablejs a lot in the past